src/components/Modals/ContactCardModal.jsx

```diff
@@ -35,29 +35,65 @@ ContactCardMenu.propTypes = {
   }).isRequired
 };
 
-const ContactCardModal = (
-  { onClose, contact, deleteContact, onDeleteContact },
-  { t }
-) => (
-  <Modal into="body" dismissAction={onClose} size="xlarge">
-    <ContactCard
-      title={t("contact_info")}
-      contact={contact}
-      renderHeader={children => (
-        <ModalHeader className="contact-card-modal__header">
-          {children}
-          <ContactCardMenu
-            deleteAction={{
-              label: t("delete"),
-              action: () => deleteContact(contact).then(() => onDeleteContact())
-            }}
+class ContactCardModal extends React.Component {
+  state = {
+    showConfirmDeleteModal: false
+  };
+
+  toggleConfirmDeleteModal = () => {
+    this.setState(state => ({
+      ...state,
+      showConfirmDeleteModal: !state.showConfirmDeleteModal
+    }));
+  };
+
+  deleteContact = async () => {
+    const { contact, deleteContact, onDeleteContact } = this.props;
+    await deleteContact(contact);
+    onDeleteContact(contact);
+  };
+
+  render() {
+    const { onClose, contact } = this.props;
+    const { showConfirmDeleteModal } = this.state;
+    const { t } = this.context;
+
+    return (
+      <Modal into="body" dismissAction={onClose} size="xlarge">
+        <ContactCard
+          title={t("contact_info")}
+          contact={contact}
+          renderHeader={children => (
+            <ModalHeader className="contact-card-modal__header">
+              {children}
+              <ContactCardMenu
+                deleteAction={{
+                  label: t("delete"),
+                  action: this.toggleConfirmDeleteModal
+                }}
+              />
+            </ModalHeader>
+          )}
+          renderBody={children => <ModalContent>{children}</ModalContent>}
+        />
+        {showConfirmDeleteModal && (
```

Comment:
If `isConfirmDelete` then render `<Modal />`. I think naming a Boolean as a condition or a question is easier to read: _isSomethingGood_, _shouldBeGood_, etc. What do you think?

Reply:
In my head, I read it as *show confirm delete modal ?*, coupled with the fact that it's clearly a boolean. But I don't mind changing it if it helps other people!